int64 index 
---
bool isDoneExecutingCommand